src/draft.rs

```diff
@@ -0,0 +1,51 @@
+use crate::*;
+
+pub type DraftGroupIndex = u32;
+pub type DraftIndex = u32;
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+#[cfg_attr(not(target_arch = "wasm32"), derive(Clone))]
+#[serde(crate = "near_sdk::serde")]
+pub struct Draft {
+    pub draft_group_id: DraftGroupIndex,
+    pub lockup_id: Option<LockupIndex>,
+    pub lockup: Lockup,
+}
+
+impl Draft {
+    pub fn assert_new_valid(&self) {
+        assert_eq!(self.lockup_id, None, "new draft must not have lockup_id");
+
+        let amount = self.lockup.schedule.total_balance();
+        self.lockup.assert_new_valid(amount);
+    }
+
+    pub fn assert_can_convert(&self) {
+        assert!(self.lockup_id.is_none(), "draft already converted");
+    }
+}
```

Comment:
1. I suggest using Rust enums to avoid impossible state to be reachable and avoid having two copies of the Lockup in the state:

```rust
enum DraftOrConvertedLockup {
    Draft(Lockup),
    Converted(LockupIndex),
}
```

This way you won't need to check on lockup_id, and you won't be able to forget to do it.

2. Why do we even need to store the converted lockups? It seems that we never remove the drafts and draft groups. I don't think it makes sense to keep drafts in the storage forever. In order to list history of lockups we planned to use Indexer (which will also be able to attach timestamp, etc)